### PR TITLE
SPORTS-196 SPORTS-210 Implement CRUD for categories

### PR DESCRIPTION
**Pull request by sanjeevkpandit**
- Implement CRUD for categories on both backend and UI.

DeepinScreenshot_20190906214219

### CODE REVIEW

File: admin-app/src/services/categories.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  const query = `
+    query {
+      ${queryAPI} {
+        id,
+        name
+      }
+    }
+  `;
+
+  return await httpHelper.getResponse(queryAPI, query);
```
Comment (SAGARACH65) [inline admin-app/src/services/categories.ts]: We don't need to await on a return statement. We can just await at the place where this function is called.

File: admin-app/src/services/categories.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  const mutation = `
+    mutation {
+      ${queryAPI} (name: "${payload.name}") {
+        id,
+        name
+      }
+    }
+  `;
+
+  return await httpHelper.getResponse(queryAPI, mutation);
```
Comment (SAGARACH65) [inline admin-app/src/services/categories.ts]: .

File: admin-app/src/services/categories.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+  const mutation = `
+    mutation {
+      ${queryAPI} (id: ${payload.id}) {
+        message
+      }
+    }
+  `;
+
+  return await httpHelper.getResponse(queryAPI, mutation);
```
Comment (SAGARACH65) [inline admin-app/src/services/categories.ts]: .

File: api/src/mutations/category.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+/**
+ * Mutation for editing an existing category by id.
+ *
+ * @export
+ * @param {*} parent
+ * @param {CategoryPayload} payload
+ * @param {Context} context
+ * @returns {Promise<object>}
+ */
```
Comment (SAGARACH65) [inline api/src/mutations/category.ts]: Looks like we are missing a `throws` block.

File: api/src/mutations/category.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+/**
+ * Mutation to delete an existing category.
+ *
+ * @export
+ * @param {*} parent
+ * @param {IdPayload} payload
+ * @param {Context} context
+ * @returns {Promise<object>}
+ */
```
Comment (SAGARACH65) [inline api/src/mutations/category.ts]: .

File: api/src/mutations/category.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
+import HttpStatus from 'http-status-codes';
+import { ApolloError } from 'apollo-server-express';
+
+import Context from '../models/Context';
+import Category from '../models/Category';
+import { IdPayload } from '../domains/general';
```
Comment (SAGARACH65) [inline api/src/mutations/category.ts]:
```suggestion
import { IDPayload } from '../domains/general';
```

File: api/src/mutations/category.ts (hunk 4 above)
Comment (sanjeevkpandit) [reply api/src/mutations/category.ts]: It's auto-generated and does not cover the @throws part.

**Pull request merged**